lightwood/api/data_source.py

```diff
@@ -305,6 +316,22 @@ def _prepare_column_encoder(self, config, is_target=False, training_data=None):
     def out_types(self):
         return [feature['type'] for feature in self.config['output_features']]
 
+    @property
+    def output_feature_names(self):
+        return [feature['name'] for feature in self.config['output_features']]
+
+    @property
+    def input_feature_names(self):
+        return [feature['name'] for feature in self.config['input_features']]
+
+    @property
+    def output_features(self):
+        return self.config['output_features']
+
+    @property
+    def input_features(self):
+        return self.config['input_features']
```

Comment:
Please just define these as getters, i.e. `input_features`.

Usually I want to avoid using @property for anything *but* external interfaces (where I use it a few times for backwards compatibility).

If the values are determined at construction time, set them in the constructor as `self.output_feature_names`... etc.

If they aren't, these should be normal getter methods.

Unless there's a reason why these have to be the way they are.

Reply:
I see now that in `fit_datasource` we were previously creating these properties, so doing this change (replacing them with getters) might require changing things in a few more places. Still, my original comment stands, but if you think this can be useful then please make a case.

Usually I'd want to introduce as few concepts in the codebase as possible, and `@property` is something I've managed to avoid until now outside of cases where it was really necessary.